### ISSUE TITLE
unable to override when starting training

### ISSUE THREAD

**Issue by snamaskar** [lbfs/DeepFaceLab_Linux#20]
[r] : rename
[d] : delete

[0] : 192HD - latest
 : 
0
Loading 192HD_SAEHD model...

Choose one or several GPU idxs (separated by comma).

[CPU] : CPU
  [0] : GeForce GTX 1080 Ti
  [1] : GeForce GTX 1070

[0] Which GPU indexes to choose? : 0,1
0,1

Press enter in 2 seconds to override model settings.
[1] Autobackup every N hour ( 0..24 ?:help ) : 1
[n] Write preview history ( y/n ?:help ) : n
[1000000] Target iteration : 1000000
[y] Flip faces randomly ( y/n ?:help ) : y
[8] Batch_size ( ?:help ) : 8
[y] Learn mask ( y/n ?:help ) : y
[n] Use learning rate dropout ( y/n ?:help ) : n
[y] Enable random warp of samples ( y/n ?:help ) : y
[0.0] GAN power ( 0.0 .. 10.0 ?:help ) : 0.0
[0.0] 'True face' power. ( 0.0000 .. 1.0 ?:help ) : 0.0
[0.0] Face style power ( 0.0..100.0 ?:help ) : 0.0
[0.0] Background style power ( 0.0..100.0 ?:help ) : 0.0
[none] Color transfer for src faceset ( none/rct/lct/mkl/idt/sot ?:help ) : none
[n] Enable gradient clipping ( y/n ?:help ) : n
[n] Enable pretraining mode ( y/n ?:help ) : n
THIS IS NOT TECH SUPPORT FOR NEWBIE FAKERS
POST ONLY ISSUES RELATED TO BUGS OR CODE

## Expected behavior

on entering within 2 seconds instead of allowing to change the options it is just displaying the above fields and starts training
## Actual behavior

*Describe, in some detail, what the program does instead. Be sure to include any error message or screenshots.*

## Steps to reproduce

*Describe, in some detail, the steps you tried that resulted in the behavior described above.*

## Other relevant information
 .
- **Operating system and version:**  linux mint using the prebuilt scripts

**Comment by snamaskar** [comment]
downloaded the fix
https://github.com/iperov/DeepFaceLab/pull/585/files

**Comment by lbfs** [comment]
I will be looking into this over the weekend. I'm also not sure if that fix is the best solution.

**Comment by iperov** [comment]
problem with the fix above is dupe stdin.fileno()

```
>>> import sys
>>> import os
>>> sys.stdin.fileno()
0
>>> os.dup( sys.stdin.fileno() )
3
```

can the issue be fixed without duping the descriptor?

is there actually any problem when opened stdin file handle will refer to duped fileno ?

**Comment by lbfs** [comment]
I'm not sure as I haven't investigated the issue, Linux I don't think there will be issues when duping the descriptor, but I agree in that I don't think it's the best solution. I want to find one that works between Windows and Linux

**Comment by iperov** [comment]
yes all work fine without dup on Windows and Colab

**Comment by lbfs** [comment]
I don't see a way to avoid using os.dup(), all of my research keeps leading back to it.

**Comment by iperov** [comment]
what about this
```
old_stdin = sys.stdin
sys.stdin = os.fdopen( os.dup(sys.stdin.fileno()) )
old_stdin.close()
```
at least old handle is closed, thus no handles leakage.

can you test it in linux

**Comment by lbfs** [comment]
That works on my machine. Allows me to properly input model parameters again during override.

**Comment by iperov** [comment]
ok I pushed the code.

**Comment by lbfs** [comment]
Alright, cool. I'll go ahead and close this issue then.
